processors/record_linker/processor.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import uuid
+import logging
+from .. import base
+logger = logging.getLogger(__name__)
+
+
+# Module API
+
+def process(conf, conn):
+
+    # Trial m2m
+    TRIAL_TABLES = [
+        'trials_conditions',
+        'trials_interventions',
+        'trials_locations',
+        'trials_organisations',
+        'trials_persons',
+        'trials_publications',
+    ]
+
+    # Unlink wrong linked records
+    # [will be reprocessed on next processors run]
+    count = 0
+    for record in base.helpers.iter_rows(conn, 'database', 'records', orderby='id'):
+        # Already unlinked
+        if not record['trial_id']:
+            continue
+        trial_records = list(conn['database']['records'].find(trial_id=record['trial_id']))
+        # Record not linked to others
+        if len(trial_records) == 1:
+            continue
+        linked_identifiers = set()
+        for trial_record in trial_records:
+            if record['id'] == uuid.UUID(trial_record['id']).hex:
+                continue
+            linked_identifiers.update(trial_record['identifiers'].items())
+        if not linked_identifiers.intersection(record['identifiers'].items()):
+            conn['database']['records'].update(
+                {'id': record['id'], 'trial_id': None}, keys=['id'])
+            logger.info('Unlinked record: %s', record['identifiers'])
+            count += 1
+        if count and not count % 100:
+            logger.info('Unlinked %s wrong linked records', count)
+    logger.info('Unlinked %s wrong linked records', count)
+
+    # Remove trials without records
+    count = 0
+    for trial in base.helpers.iter_rows(conn, 'database', 'trials', orderby='id'):
```

Comment:
No need for `orderby` here, and also I think you can add `conn['database']['trials'].columns.source_id != 'pubmed'`, so could remove lines 55 and 56.

Actually, considering you're looking for trials without records, I would do an actual query like:

`SELECT trials.id FROM trials LEFT JOIN records ON records.trial_id = trials.id WHERE records.trial_id IS NULL AND trials.source_id != 'pubmed'`

This makes this code much quicker, as there's only a single request needed and we're only getting the trial id, and not every column. On the other hand, it's loading all IDs in memory instead of iterating over them. This shouldn't be a problem, though, as considering that one UUID has only 16 bytes, even a million UUIDs will require only 16 MB of memory (disconsidering possible Python/Dataset overheads).